pages/docs/coding-conventions/testing.en-US.mdx

```diff
@@ -18,30 +19,41 @@ The following are some general guidelines for testing:
   ```tsx
   // ❌
   const submitButton = await waitFor(() =>
-    screen.getByRole('button', { name: /submit/i }),
-  )
+    screen.getByRole("button", { name: /submit/i })
+  );
 
   // ✅
-  const submitButton = await screen.findByRole('button', { name: /submit/i })
+  const submitButton = await screen.findByRole("button", { name: /submit/i });
   ```
 
   - Don't pass an empty callback to `waitFor`.
 
   ```tsx
   // ❌
-  await waitFor(() => {})
-  expect(window.fetch).toHaveBeenCalledWith('foo')
-  expect(window.fetch).toHaveBeenCalledTimes(1)
+  await waitFor(() => {});
+  expect(window.fetch).toHaveBeenCalledWith("foo");
+  expect(window.fetch).toHaveBeenCalledTimes(1);
 
   // ✅
-  await waitFor(() => expect(window.fetch).toHaveBeenCalledWith('foo'))
-  expect(window.fetch).toHaveBeenCalledTimes(1)
+  await waitFor(() => expect(window.fetch).toHaveBeenCalledWith("foo"));
+  expect(window.fetch).toHaveBeenCalledTimes(1);
   ```
 
   - Don't add `aria-*`, `role`, and other accessibility attributes to elements unless the application actually uses them.
   - Don't use Testing Library's ESLint plugins: [eslint-plugin-testing-library](https://github.com/testing-library/eslint-plugin-testing-library) and [eslint-plugin-jest-dom](https://github.com/testing-library/eslint-plugin-jest-dom).
   - Don't use the `cleanup` function. RTL takes care of cleaning up after each test.
-  - Don't use `screen` to query for elements. Use `screen` for querying elements and debugging assertions.
+  - Don't use `screen` on its own to assert for elements, wrap it in `expect` instead. If you want to assert that something exists, make that assertion explicit.
+
+    ```tsx
+    // ❌
+    screen.getByRole("button", { name: /dispense/i });
+
+    // ✅
+    expect(
+      screen.getByRole("button", { name: /dispense/i })
+    ).toBeInTheDocument();
+    ```
```

Comment:
Reading this back, it was originally meant to explain this point from Kent's blog:

![CleanShot 2024-12-03 at 1  55 02@2x](https://github.com/user-attachments/assets/ad9dbe6e-1059-4a08-b462-403dccc6d347)

For our purposes, we should probably amend it to read something like:

```mdx
  - Use `screen` to query and debug elements. This is the recommended way to find elements in your tests as it ensures you're working with the current document and provides better error messages.

    ```tsx
    // ❌
    const { getByText } = render(<MyComponent />)
    const element = getByText(/hello/i)

    // ✅
    render(<MyComponent />)
    const element = screen.getByText(/hello/i)
    ```
```